### PR TITLE
Refactoring client

### PR DESCRIPTION
**Pull request by bpb27**

### CODE REVIEW

File: theoktany/__init__.py [hunk 1]
```diff
@@ -1,4 +0,0 @@
-from theoktany.auth_client import *
```
Comment (stevenhair) [inline theoktany/__init__.py]: Why did you remove these? Just curious.

File: theoktany/validate.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+def validate(response, status_code):
```
Comment (bpb27) [inline theoktany/validate.py]: for some reason validate wasn't importing from the __init__.py file, so i pulled it out...

File: theoktany/__init__.py (hunk 1 above)
Comment (bpb27) [inline theoktany/__init__.py]: were they doing anything?

Comment (stevenhair) [inline theoktany/__init__.py]: They weren't importing like `from theoktany import ApiClient`? Strange.

**Comment by stevenhair** [comment]
It looks like there's a user dictionary that's being passed around. I thought that we concluded that we should just use method arguments?

**Comment by bpb27** [comment]
that's true. i'll make the change.

File: tests/test_auth_client/stubs/test_send_sms_challenge.json [hunk 3]
```diff
@@ -16,53 +16,16 @@
...
-                }
-              },
-              "user": {
-                "href": "https://your-domain.okta.com/api/v1/users/00u15s1KDETTQMQYABRL",
-                "hints": {
-                  "allow": [
-                    "GET"
-                  ]
-                }
+              "id": "sms00001",
```
Comment (tsmarsh) [inline tests/test_auth_client/stubs/test_send_sms_challenge.json]: This really gets that much simpler? Neat!

File: tests/test_auth_client/test_sms_auth.py [hunk 4]
```diff
@@ -1,210 +1,139 @@
...
+user_two = dict(id="00002", login="bobby@aol.com", phone_number="+1-555-555-5555")
+user_three = dict(id="00003", login="freddy@aol.com", phone_number="+1-555-555-5555")
+
 
 class SMSAuthTests(unittest.TestCase):
 
     def setUp(self):
-        self.api_client = ApiClient()
-        self.auth_client = OktaAuthClient(self.api_client)
+        self.api_client = ApiClient(API_TOKEN='jrr-tolkien')
```
Comment (tsmarsh) [inline tests/test_auth_client/test_sms_auth.py]: nice
